src/modelfree/policies/wrappers.py

```diff
@@ -36,3 +36,128 @@ def predict(self, observation, state=None, mask=None, deterministic=False):
+                # with the shape that the policy expects. Inferred state shapes will be set
+                # for stateful policies as soon as they return a state vector, i.e. after the
+                # first prediction step
+                standardized_state = self._standardize_state(state, env_mask,
+                                                             self.inferred_state_shapes[i])
+
+            predicted_actions, new_states = policy.predict(observation,
+                                                           state=standardized_state,
+                                                           mask=mask,
+                                                           deterministic=deterministic)
+            if new_states is not None and self.inferred_state_shapes[i] is None:
+                # If this is a policy that returns state, and its current inferred state
+                # is None, update the inferred state value to this shape
+                self.inferred_state_shapes[i] = new_states.shape
+
+            # This is basically numpy mask value assignment, but created to work with python
+            # arrays to accommodate the fact that different policies won't have state
+            # values of the same shape/type
+            policy_actions = self._array_mask_assign(policy_actions, env_mask, predicted_actions)
+            new_state_array = self._array_mask_assign(new_state_array, env_mask, new_states)
+
+        return policy_actions, new_state_array
+
+    def _standardize_state(self, state_arr, env_mask, inferred_state_shape):
+        """
+        Solves the problem of different policies taking in different types of state vector:
+        either different shapes of array, or None in the case of a MLP policy. Takes all the
+        entries of state_arr that are true in env_mask
+        """
+        env_mask = env_mask.copy()
+        if inferred_state_shape is None:
+            filler_value = None
+        else:
+            filler_value = np.zeros(shape=inferred_state_shape)
+            for i in range(len(state_arr)):
+                if env_mask[i] and state_arr[i] is None:
+                    # This is to account for the edge case where we are sending this state to a
+                    # stateful policy, but at the last step this env had a stateless policy,
+                    # and thus had its state value set to None. Stateful policies can take in
+                    # None as a state value, but it can't accommodate some of its envs having
+                    # arrays and others being None
+                    env_mask[i] = False
+
+        # Fill in `filler_value` for all indices of state_arr where env_mask is False
+        filled_state = self._array_mask_assign(state_arr,
+                                               [not el for el in env_mask],
+                                               filler_value)
+        return filled_state
+
+    @staticmethod
```

Comment:
Huh, I feel like staticmethods served a useful purpose, in that it now does seem pretty organizationally strange to me to have methods that are very bound up in the logic of this specific class definition defined in the broader module (especially since _standardize_state also doesn't use the self object, so based on your objection to staticmethods I also made it module level). But I'm fine going along with the style decision.